syntax = "proto3";

import public "google/protobuf/empty.proto";

import public "account.proto";
import public "customer.proto";

package bank;

service BankService {
	rpc signUp (bank.customer.SignUpRequest) returns (bank.customer.SignUpResponse) {}
	rpc Login (bank.customer.LogInRequest) returns (bank.customer.LogInResponse) {}

	rpc CreateFixedDepositAccount(bank.account.CreateFixedDepositAccountRequest) returns (bank.account.CreateFixedDepositAccountResponse) {}
	rpc CreateNormalSavingsAccount(bank.account.CreateNormalSavingsAccountRequest) returns (bank.account.CreateNormalSavingsAccountResponse) {}

	rpc ListTransactionLogsByAccNum(bank.account.ListTransactionLogsRequest) returns (bank.account.ListTransactionLogsResponse) {}
	rpc ListAccountsByUserId(bank.account.ListAccountsByUserIdRequest) returns (bank.account.ListAccountsByUserIdResponse) {}

	rpc Transfer(bank.account.TransferRequest) returns (bank.account.TransferResponse) {}

}

//response의 1번은 api의 성공과 실패 여부를 반환합니다.
//response의 2번은 status가 false일 경우 에러 메시지 또는 에러 코드를 반환합니다.